Sources/Testing/Running/Runner.swift

```diff
@@ -209,11 +209,33 @@ extension Runner {
             if let testCases = step.test.testCases {
               try await _runTestCases(testCases, within: step)
             }
+
+            // Run the children of this test (i.e. the tests in this suite.)
+            try await _runChildren(of: stepGraph, depth: depth, lastAncestorStep: lastAncestorStep)
           }
         }
       }
+    } else {
```

Comment:
This `else` is for this line above, right?
```
if let step = stepGraph.value, case .run = step.action {
```
Isn't it possible that `value` is non-`nil` but `step.action` is not `.run`? In which case, we should not continue to run the children, right?

Reply:
No, I think we still want to traverse them in order to indicate we've skipped them, for instance (remember that the action is propagated downward during planning.)

This isn't a regression at any rate, so if we want to rethink that, let's do so in a separate PR dedicated to it?

Reply:
Ah right — the function name "**run**Children(...)" threw me off. I recall now it doesn't necessarily _run_ them, but more like "evaluate" or "process" them, depending on their action. Sounds good